.github/workflows/hosted.yml

```diff
@@ -63,16 +63,36 @@ jobs:
       with:
         maven-version: 3.9.4
 
-    - name: Build ANTLR with Maven
+    - name: Build ANTLR modules without testing
       run: mvn install -DskipTests=true -Darguments="-Dmaven.javadoc.skip=true" -B -V
 
+
+    - name: Unit test core module
+      if: matrix.target != 'tool'
+      run: |
+        cd runtime/Core
+        mvn test
+
+    - id: capitalize
+      name: Capitalize target module name
+      if: matrix.target != 'tool'
+      uses: ASzc/change-string-case-action@v6
+      with:
+        string: ${{ matrix.target }}
+    
+    - name: Unit test target module
+      if: matrix.target != 'tool'
+      run: |
+        cd runtime/${{ steps.capitalize.outputs.capitalized }}
+        mvn test
+
     - name: Test tool
       if: matrix.target == 'tool'
       run: |
         cd tool-testsuite
         mvn test
 
-    - name: Test runtime (Windows)
+    - name: End-to-end test runtime (Windows)
```

Comment:
What does `End-to-end` mean? Why just `Test` is not enough?

Reply:
Because it's different from unit tests, which are local to a module. This is testing complete tool chain: tool, target and runtime